compass/landice/tests/slm/circ_icesheet/setup_mesh.py

```diff
@@ -0,0 +1,409 @@
+import os
+import shutil
+
+import mpas_tools.io
+import netCDF4
+import numpy as np
+from mpas_tools.io import write_netcdf
+from mpas_tools.logging import check_call
+from mpas_tools.mesh.conversion import cull
+from mpas_tools.planar_hex import make_planar_hex_mesh
+from mpas_tools.scrip.from_mpas import scrip_from_mpas
+from mpas_tools.translate import center, translate
+from netCDF4 import Dataset as NetCDFFile
+
+from compass.model import make_graph_file
+from compass.step import Step
+
+
+class SetupMesh(Step):
+    """
+    A step for creating a mesh with initial conditions plus forcing file for
+    circular icesheet test
+
+    Attributes
+    ----------
+    res : str
+        Resolution of MALI mesh
+
+    nglv : str
+        Number of Gauss-Legendre nodes in latitude in the SLM
+    """
+
+    def __init__(self, test_case, name, res, nglv):
+        """
+        Create the step
+
+        Parameters
+        ----------
+        test_case : compass.TestCase
+            The test case this step belongs to
+
+        res : str
+            Resolution of MALI mesh
+
+        nglv : str
+            Number of Gauss-Legendre nodes in latitude in the SLM
+        """
+        super().__init__(test_case=test_case, name=name)
+
+        self.add_output_file(filename='graph.info')
+        self.add_output_file(filename='landice_grid.nc')
+
+        self.res = res
+        self.nglv = nglv
+    # no setup() method is needed
+
+    def run(self):
+        """
+        Run this step of the test case
+        """
+        logger = self.logger
+        config = self.config
+        section = config['circ_icesheet']
+
+        lx = section.getfloat('lx')
+        ly = section.getfloat('ly')
+        dc = float(self.res) * 1000.0
+
+        # calculate the number of grid cells in x and y direction
+        # for the uniform, hexagonal planar mesh
+        nx = max(2 * int(0.5 * lx / dc + 0.5), 4)
+        # factor of 2/sqrt(3) because of hexagonal mesh
+        ny = max(2 * int(0.5 * ly * (2. / np.sqrt(3)) / dc + 0.5), 4)
+
+        mpas_tools.io.default_format = 'NETCDF4'
+        mpas_tools.io.default_engine = 'netcdf4'
+
+        # call the mesh creation function
+        dsMesh = make_planar_hex_mesh(nx=nx, ny=ny, dc=dc,
+                                      nonperiodic_x=True,
+                                      nonperiodic_y=True)
+        dsMesh = cull(dsMesh, logger=logger)
+        # adding the time dimension is needed for netcdf4 formatting to work
+        dsMesh['xtime'] = ('Time', ['2015-01-01_00:00:00'.ljust(64)])
+        # translating the mesh center to x=0 & y=0
+        center(dsMesh)
+        # shift the center to a quarter or radius
+        shift = 200000.0
```

Comment:
Do you want to make this a cfg option?  It's fine if you want to leave it for now, and we can always add that option later if we need it.  When you/we write the docs, we should make sure to address why this is being done.